src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -545,6 +545,51 @@ internal string GetUIVisualizerName(string serviceId, int id)
 
         private delegate IVariableInformation Traverse(IVariableInformation node);
 
+        // new function here
+        private string GetDisplayNameFromArrayIndex(uint arrayIndex, int rank, uint[] dimensions, bool isForward)
```

Comment:
>GetDisplayNameFromArrayIndex [](http://example.com/codeflow?start=23&length=28)

nit: I would suggest moving this function lower down in the file. I find code easier to read if the entry points to the class are near the top and the private helpers are near the bottom roughly sorted so that the callers are above the callees. That way you can read a file without too much jumping around.